generic3g/MAPL_Generic.F90

```diff
@@ -575,26 +598,85 @@ subroutine gridcomp_connect_all(gridcomp, src_comp, dst_comp, rc)
       _RETURN(_SUCCESS)
    end subroutine gridcomp_connect_all
 
-   subroutine hconfig_get_string(hconfig, keystring, value, default, rc)
-      type(ESMF_HConfig), intent(inout) :: hconfig
-      character(*), intent(in) :: keystring
-      character(:), allocatable :: value
-      character(*), optional, intent(in) :: default
+   subroutine gridcomp_get_hconfig(gridcomp, hconfig, rc)
+      type(ESMF_GridComp), intent(inout) :: gridcomp
+      type(ESMF_HConfig), intent(out) :: hconfig
       integer, optional, intent(out) :: rc
 
       integer :: status
-      logical :: has_key
+      type(ESMF_Config) :: config
+      
+      call ESMF_GridCompGet(gridcomp, config=config, _RC)
+      call ESMF_ConfigGet(config, hconfig=hconfig, _RC)
```

Comment:
This utility will force me (us) to resolve a problem in the existing implementation.   OuterMeta also has-a hconfig object.   I'm not yet confident that it is the same reference as one obtains with the calls you provided.    We'll discuss - no change for now.
```suggestion
      call ESMF_GridCompGet(gridcomp, config=config, _RC)
      call ESMF_ConfigGet(config, hconfig=hconfig, _RC)
```